k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -2982,10 +2982,30 @@ public void onMessageBuildSuccess(MimeMessage message, boolean isDraft) {
             currentMessageBuilder = null;
             new SendMessageTask(getApplicationContext(), mAccount, mContacts, message,
                     mDraftId != INVALID_DRAFT_ID ? mDraftId : null).execute();
+            updateReferencedMessage();
```

Comment:
This triggers database operations in the calling thread. We should probably call it from `SendMessageTask`.